flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/StreamTaskTimerTest.java

```diff
@@ -107,43 +109,42 @@ public void testErrorReporting() throws Exception {
 
     @Test
     public void checkScheduledTimestamps() throws Exception {
-        final AtomicReference<Throwable> errorRef = new AtomicReference<>();
-
-        final long t1 = System.currentTimeMillis();
-        final long t2 = System.currentTimeMillis() - 200;
-        final long t3 = System.currentTimeMillis() + 100;
-        final long t4 = System.currentTimeMillis() + 200;
-
-        timeService.registerTimer(t1, new ValidatingProcessingTimeCallback(errorRef, t1, 0));
-        timeService.registerTimer(t2, new ValidatingProcessingTimeCallback(errorRef, t2, 1));
-        timeService.registerTimer(t3, new ValidatingProcessingTimeCallback(errorRef, t3, 2));
-        timeService.registerTimer(t4, new ValidatingProcessingTimeCallback(errorRef, t4, 3));
-
-        long deadline = System.currentTimeMillis() + 20000;
-        while (errorRef.get() == null
-                && ValidatingProcessingTimeCallback.numInSequence < 4
-                && System.currentTimeMillis() < deadline) {
-            Thread.sleep(100);
+        ValidatingProcessingTimeCallback.numInSequence = 0;
+        long currentTimeMillis = System.currentTimeMillis();
+        ArrayList<ValidatingProcessingTimeCallback> timeCallbacks = new ArrayList<>();
+
+        /*
+         It is not possible to register timer for currentTimeMillis or value slightly greater than
```

Comment:
I believe that instead of:
> It is not possible to register timer for currentTimeMillis

You had this in mind?

> It is not possible to test registering timer for currentTimeMillis

?